pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionRuntimeManager.java

```diff
@@ -637,11 +628,11 @@ private void updateAssignment(Assignment assignment) {
                             FunctionRuntimeInfo newFunctionRuntimeInfo = new FunctionRuntimeInfo();
                             newFunctionRuntimeInfo.setFunctionInstance(assignment.getInstance());
 
-                            this.insertStartAction(newFunctionRuntimeInfo);
-                            this.setFunctionRuntimeInfo(fullyQualifiedInstanceId, newFunctionRuntimeInfo);
+                            this.conditionallyStartFunction(newFunctionRuntimeInfo);
+                            this.functionRuntimeInfoMap.put(fullyQualifiedInstanceId, newFunctionRuntimeInfo);
                         }
                     } else {
-                        deleteFunctionRuntimeInfo(fullyQualifiedInstanceId);
```

Comment:
since we are now setting and deleting items from functionRuntimeInfoMap during initialization, can we add a new test or add to the existing "testRuntimeManagerInitialize" test to make sure that functionRuntimeInfoMap has the correct functionRuntimeInfos and is in-sync with the assignments the worker should have after initialization.

Reply:
Added